### PR TITLE
conditional requests guidelines

### PR DESCRIPTION
**Pull request by bterlson**
This codifies what we've been doing in JS which is a little different because sometimes we have a model type with `etag` on it, but sometimes we don't, in which case the user needs a way to just give us the etag value.

### CODE REVIEW

File: docs/typescript/design.md [hunk 1]
```diff
@@ -606,7 +606,23 @@ Polling configuration may be used only in the absence of relevant retry-after he
 
 TODO: If this is largely implemented for the API Designer, please include an example of how to use the Azure Core type in the public API.  It would be ideal to remove guidelines where the requirement has already been addressed for the API Designer in the type.
 
-TODO: Please add a section on conditional request methods.
+## Conditional Request Methods {#ts-conditional-requests}
+
+There are two patterns in use depending on whether `etag` is a member of the conceptual entity or not.
```
Comment (annelo-msft) [inline docs/typescript/design.md:611]: Looks like you reverted to conceptual entity in the second commit ...

File: docs/typescript/design.md [hunk 2]
```diff
@@ -606,7 +606,23 @@ Polling configuration may be used only in the absence of relevant retry-after he
...
+* onlyIfUnchanged - sets the `if-none-match` header to the `etag`.
+* onlyIfMissing - sets the `if-none-match` header to `*`.
+* onlyIfPresent - sets the `if-match` header to `*`.
+
+{% include requirement/MUST id="ts-conditional-request-options-2" %} provide the following options in a method's options bag's `conditions` property when the conceptual entity does not have an `etag` property:
+
+* ifMatch - sets the `if-match` header to the value provided.
+* ifNoneMatch - sets the `if-none-match` header to the value provided.
+* ifModifiedSince - sets the `if-modified-since` header to the value provided
+* ifUnmodifiedSince - sets the `if-unmodified-since` header to the value provided.
```
Comment (annelo-msft) [inline docs/typescript/design.md:625]: If the API designer chooses to use both the convenience property (case 1) and the `conditions` property (case 2), and a developer passes in both, what should the behavior be?

Comment (bterlson) [reply docs/typescript/design.md:625]: It hasn't come up yet, but I'll put in what we discussed last night - that it's just an error.

**Review by annelo-msft — APPROVED** [review]

**Review by xirzec — APPROVED** [review]
I'm getting excited by all these guidelines updates! Nice work! 👍

**Pull request merged**